File: backend/src/main/java/group8/bloodbank/model/Item.java
package group8.bloodbank.model; /***********************************************************************
 * Module:  Item.java
 * Author:  david
 * Purpose: Defines the Class Item
 ***********************************************************************/

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

/**
 * @pdOid 67762e00-bfac-4da1-b53c-d466a4467f6e
 */
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String name;
    @Column
    private int quantity;
    @ManyToOne
    @JoinColumn(name = "blood_bank_id")
    private BloodBank bloodBank;

}

File: frontend/src/app/model/item.ts
import { BloodBank } from "./bloodBank";


export class Item {
    id!: number;
    bloodBank!: BloodBank;
    quantity!: number;
    name!: string;

    constructor(obj?: any){
        if(obj) {
            this.id = obj.id;
            this.bloodBank = obj.bloodBank;
            this.quantity = obj.quantity;
            this.name =  obj.name;
        }
    }
}